### PR TITLE
bot task reminder nodejs

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**
This sample shows feature where user can schedule a task and get reminder on the scheduled date and time.

### CODE REVIEW

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    npm start
+    ```
+- **Manually update the manifest.json**
+    - Edit the `manifest.json` contained in the  `appPackage/` folder to replace with your MicrosoftAppId (that was created in step1.1 and is the same value of MicrosoftAppId in `.env` file) *everywhere* you see the place holder string `{MicrosoftAppId}` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
+
+## Features of this sample
+
+- Card with actions `Schedule task`.
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:61]: This is wrong, action name is "create-reminder"

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+- **Manually update the manifest.json**
+    - Edit the `manifest.json` contained in the  `appPackage/` folder to replace with your MicrosoftAppId (that was created in step1.1 and is the same value of MicrosoftAppId in `.env` file) *everywhere* you see the place holder string `{MicrosoftAppId}` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
+
+## Features of this sample
+
+- Card with actions `Schedule task`. 
+
+![Schedule task ](Images/ScheduleTaskCard.png)
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:63]: Contents are wrong:
"Please click here to schedule a recurring task reminder"

image

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+![Task Details ](Images/ScheduleTask.png)
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:65]: This image is wrong:
1) Title is wrong
2) Subtitle should be "Schedula a task"
3) Task Title should be "Task title"
image

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+- Task reminder
+
+![Task reminder](Images/TaskReminder.png)
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:69]: Content are wrong:
1) Please click here to schedula a recurring task
2) **"Task submitted successfully, you will get a recurring reminder for the task at a scheduled time"**
3) **"Reminder for a scheduled task!"**
4) There is no space after colon (:)
image

File: samples/bot-task-reminder/nodejs/appPackage/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# App Package
+
+Your "app package" is used to publish your app configuration to Teams by manually using the "Upload a custom app" from the app gallery in Teams. 
+
+This package can also be used to manually publish your app to the Teams store or your organization.
+
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/appPackage/README.md:6]: Is this file required?
If yes, please check and update the contents.

File: samples/bot-task-reminder/nodejs/appPackage/manifest.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.11/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.11",
+  "version": "1.0.1",
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/appPackage/manifest.json:4]: **It should be "version": "1.0.0"**

File: samples/bot-task-reminder/nodejs/appPackage/manifest.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    "privacyUrl": "https://www.microsoft.com/privacy",
+    "termsOfUseUrl": "https://www.microsoft.com/termsofuse"
+  },
+  "name": {
+    "short": "App task reminder",
+    "full": "App task reminder"
+  },
+  "description": {
+    "short": "App task reminder",
+    "full": "This sample shows feature where user can schedule a task and get reminder on the scheduled date and time."
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/appPackage/manifest.json:19]: This sample shows a feature where user can schedule a recurring task and get a reminder on the scheduled time

File: samples/bot-task-reminder/nodejs/appPackage/manifest.json [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      ],
+	  "commandLists": [
+        {
+          "scopes": [
+            "personal"
+          ],
+          "commands": [
+            {
+              "title": "create-reminder",
+              "description": "To create reminder for task"
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/appPackage/manifest.json:40]: To create the reminder for a task

File: samples/bot-task-reminder/nodejs/bots/teamsBot.js [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+        this.baseUrl = process.env.BaseUrl;
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (let member = 0; member < membersAdded.length; member++) {
+                if (membersAdded[member].id !== context.activity.recipient.id) {
+                    await context.sendActivity("Hello and welcome! With this sample you can schedule a task and get reminder on the scheduled date and time.(use command 'create-reminder')");
+                }
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/bots/teamsBot.js:20]: you can schedule a recurring task and get a reminder on the scheduled time

File: samples/bot-task-reminder/nodejs/views/ScheduleTask.ejs [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+                    });
+                }
+                else {
+                    $(this).css({
+                        "border": "",
+                        "background": ""
+                    });
+                }
+            });
+            if (isValid == false) {
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/views/ScheduleTask.ejs:40]: Line space before it.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed and provided my feedback.

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
+## Bot task reminder
+
+This sample shows a feature where user can schedule a recurring task and get the reminder at scheduled time.
+
+Type command create-reminder to get card for scheduling the recurring task.
+
+![Schedule task ](Images/ScheduleTaskCard.png)
+
+![Task Details ](Images/ScheduleTask.png)
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:9]: Subtitle spelling is wrong here:

image

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    npm start
+    ```
+- **Manually update the manifest.json**
+    - Edit the `manifest.json` contained in the  `appPackage/` folder to replace with your MicrosoftAppId (that was created in step1.1 and is the same value of MicrosoftAppId in `.env` file) *everywhere* you see the place holder string `{MicrosoftAppId}` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
+
+## Features of this sample
+
+- Card with actions `Schedule task`.
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:63]: action

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
+
+## Features of this sample
+
+- Card with actions `Schedule task`. 
+
+![Schedule task ](Images/ScheduleTaskCard.png)
+
+![Task Details ](Images/ScheduleTask.png)
```
Comment (v-chetsh) [inline samples/bot-task-reminder/nodejs/Readme.md:67]: Task module subtitle spelling is wrong here:
image

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed and provided my feedback.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Provided feedback

File: samples/bot-task-reminder/nodejs/Readme.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+2) In the folder where repository is cloned navigate to `samples/bot-task-reminder/nodejs`
+
+3) Install node modules
+
+Inside node js folder, open your local terminal and run the below command to install node modules. You can do the same on Visual studio code terminal by opening the project on Visual studio code 
+
+    ```bash
+    npm install
+    ```
```
[the remark this answers is not in the document]
Comment (v-chetsh) [reply samples/bot-task-reminder/nodejs/Readme.md:45]: Fixed

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving the pr.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**